### PR TITLE
UI improvements for audio track

### LINKED ISSUE #147: [Feature] Waveform focus styles
## Problem
The waveform does not have a clear focus style, meaning that sighted keyboard users will have a difficult time knowing that they are able to seek using the keyboard like other "seek bars" that exist.

## Description
We should add a focus style that makes it clear to sighted keyboard users that they can seek using the arrow keys. Screen reader users will get an indiciation from their screen reader that they are in a "slider" which will give the hint that they can use the arrow keys to interact, but for non-screen reader keyboard users this will not apply. Therefore we need some kind of visual indication that the waveform is focused and seekable.

One potential solution is to make the vertical "seek" line a solid color with a circle or bulb at the top. Similar to how iOS represents a video scub in the Photos app, but with a bulb at the top of the line to add an extra indication (especially when the audio is at the `0:00` timestamp something will need to be visible inside of the waveform).

IMG_1064

## Alternatives
I'm sure there are other designs we could do for the focus styles, but I'm not sure of any.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #190: [Feature] Waveform keyboard seek to start/end
## Problem
Some keyboard navigable audio/video seek bars are able to be jumped to the start or end of the audio by using `cmd/super + left/right` or `home/end`. We don't currently implement this. 

## Description
Add the ability to seek to the start or end of the audio using the keyboard either through `meta + left/right` or `home/end`

## Additional context
I got this idea checking out this other media player's implementation: https://github.com/IET-OU/mediaelement/commit/25eb64a2097f25bffd5c1420807f5605cbe4e0b0#diff-219f920a90e9fec6455e5cb6863cf6e51831182cd586184b53e5ae3a4e94b32aR102-R108

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #191: [Bug] Play/pause button needs rounded corners
## Description
The play/pause button current has square corners, but should be updated to match the design and have rounded corners: https://www.figma.com/file/vEoItpMQyFZb5AZV9aoP7I/Audio-integration?node-id=1107%3A12561

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #147 by @sarayourfriend
Fixes #190 by @sarayourfriend
Fixes #191 by @sarayourfriend

## Description

This PR adds the focus indicator to the waveform.

## Known issues
The focus outline of the play/pause button interferences with the waveform seekbar, likely due to complex `z-index` layers inside the `Waveform` component. This might be tricky to solve.

## Screenshots

https://user-images.githubusercontent.com/16580576/132212606-9dd3d87a-1cd3-475e-93e7-da66b9af6a85.mov

Uses <kbd>Tab</kbd> and <kbd>Shift</kbd>+<kbd>Tab</kbd> to focus forward and backward respectively.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
Looks good. Two things:

1. My waveform has a vertical scrollbar, that is easily removed by replacing `overflow-x-hidden` on the waveform with `overflow-hidden`.
2. The play/pause overflow issue is caused becasue the waveform has relative positioning, which gives it a [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) and places it in front of the play/pause button. We can add the `relative` class to the play/pause button or any of its parents to fix that.

File: src/components/AudioTrack/Waveform.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div
     ref="el"
-    class="waveform relative bg-dark-charcoal-04 overflow-x-hidden"
+    class="waveform relative bg-dark-charcoal-04 overflow-x-hidden focus:outline-none"
```
Comment (sarayourfriend) [inline src/components/AudioTrack/Waveform.vue:4]: Will this disable the extra outline on high contrast mode on Windows? 🤔 I think we shouldn't necessarily fight the natural focus outline when it exists.

**Review by sarayourfriend — COMMENTED** [review]
https://user-images.githubusercontent.com/24264157/132225697-bdf78e64-bd64-4a80-98ee-fb44a0ef6c3d.mov

Code looks great! There a couple of issues though. In firefox, when pressing `space` to toggle the play/pause, the default behavior of the space bar activates in the waveform element and it is scrolled down. I guess there is some y overflow. This would also trigger if someone was scrolling over the page and accidentally started their scroll inside the waveform element. I guess we should ensure that overflow-y is hidden?

Secondly, (this probably warrants a separate issue/PR) I finally got to test it using VoiceOver in Safari and the entire interaction is broken. The aria label is not translated (the `$t...` is read out literally) and the `valuetext` is not correct read on interaction. Furthermore, the instructions for interacting with the element don't work correctly, `control+option+left/right` doesn't do anything and `left/right` arrows alone seem to be navigating inside the text of the timestamp rather than the manipulating the slider itself.


https://user-images.githubusercontent.com/24264157/132226792-262b451e-a499-49f4-a584-5196b1944e13.mov

To me the only blocker here though is fixing the y-overflow. The rest can be addressed in a separate PR.

File: src/components/AudioTrack/Waveform.vue (hunk 1 above)
Comment (zackkrida) [reply src/components/AudioTrack/Waveform.vue:4]: > The outline-none utility is implemented using a transparent outline under the hood to ensure elements are still visibly focused to Windows high contrast mode users.

No, thankfully! https://tailwindcss.com/docs/outline

Comment (sarayourfriend) [reply src/components/AudioTrack/Waveform.vue:4]: Right on!

**Comment by zackkrida** [comment]
@dhruvkb final focus mockup is [here](https://www.figma.com/file/vEoItpMQyFZb5AZV9aoP7I/Audio-integration?node-id=1151%3A14566):

CleanShot 2021-09-06 at 10 13 47@2x

**Comment by dhruvkb** [comment]
This looks really good! I'll mark this PR as draft and take it back to the workshop.

**Comment by sarayourfriend** [comment]
> The focus outline of the play/pause button interferences with the waveform seekbar, likely due to complex z-index layers inside the Waveform component. This might be tricky to solve.

I think this would be solved in the newest design that Zack shared above where the play/pause button is separated from the waveform, right?

**Comment by dhruvkb** [comment]
I'll work on the VoiceOver issues separately as there is a bit of reading involved and I wouldn't want to block #195 for the focus styles.

**Review by sarayourfriend — APPROVED** [review]
LGTM! Nice work 🚀

**Pull request merged**